py4DSTEM/process/utils/utils.py

```diff
@@ -607,3 +608,204 @@ def get_ewpc_filter_function(Q_Nx, Q_Ny):
+        scaling along the x,y or kx,ky axes.)
+
+    Args:
+        array (2D/4D numpy array): Input array, or 4D stack of arrays, to be resized. 
+        scale (float): scalar value giving the scaling factor for all dimensions
+        output_size (2-tuple of ints): two values giving either the (x,y) output size for 2D, or (kx,ky) for 4D
+        force_nonnegative (bool): Force all outputs to be nonnegative, after filtering
+        bandlimit_nyquist (float): Gaussian filter information limit in Nyquist units (0.5 max in both directions)
+        bandlimit_power (float): Gaussian filter power law scaling (higher is sharper)
+        dtype (numpy dtype): datatype for binned array. default is single precision float
+
+    Returns:
+        the resized array (2D/4D numpy array)
+    """
+    input__size = array.shape[-2:]
+
+
+    if scale is not None:
+        assert output_size is None, 'Cannot specify both a scaling factor and output size'
+        assert np.size(scale) == 1, 'scale should be a single value'
+        scale = np.asarray(scale)
+        output_size = (input__size * scale).astype('intp')
+    else:
+        assert scale is None, 'Cannot specify both a scaling factor and output size'
+        assert np.size(output_size) == 2, 'output_size must contain two values'
+        output_size = np.asarray(output_size)
+    
+    scale_output = np.prod(output_size) / np.prod(input__size)
+
+
+    if bandlimit_nyquist is not None:
+        if len(input__size) == 2:
+            kx = np.fft.fftfreq(output_size[0])
+            ky = np.fft.fftfreq(output_size[1])
+        if len(input__size) == 4:
+            kx = np.fft.fftfreq(output_size[2])
+            ky = np.fft.fftfreq(output_size[3])
+        k2 = kx[:,None]**2 + ky[None,:]**2
+        # Gaussian filter 
+        k_filt = np.exp((k2**(bandlimit_power/2))/(-2*bandlimit_nyquist**bandlimit_power))
+
+
+    # generate slices
+    # named as {dimension}_{corner}_{in_/out},
+    # where corner is ul, ur, ll, lr for {upper/lower}{left/right}
+
+    # x slices
+    if output_size[0] > input__size[0]:
+        # x dimension increases
+        x_ul_out = slice(0,input__size[0]//2)
+        x_ul_in_ = slice(0,input__size[0]//2)
+
+        x_ll_out = slice(0-input__size[0]//2+output_size[0], output_size[0])
+        x_ll_in_ = slice(0-input__size[0]//2+input__size[0], input__size[0])
+
+        x_ur_out = slice(0, input__size[0]//2)
+        x_ur_in_ = slice(0, input__size[0]//2)
+
+        x_lr_out = slice(0-input__size[0]//2+output_size[0], output_size[0])
+        x_lr_in_ = slice(0-input__size[0]//2+input__size[0], input__size[0])
+
+    elif output_size[0] < input__size[0]:
+        # x dimension decreases
+        x_ul_out = slice(0, output_size[0]//2)
+        x_ul_in_ = slice(0, output_size[0]//2)
+
+        x_ll_out = slice(0-output_size[0]//2, output_size[0])
+        x_ll_in_ = slice(0-output_size[0]//2+input__size[0], input__size[0])
```

Comment:
I'm not sure what this code is doing exactly yet, but symmetry considerations suggest to me that there may be something wrong here...?

Reply:
Tested and outputs seem correct!  I believe it matches my original long-form written out version.

Reply:
With further reflection, I actually think you are correct @bsavitzky !

I am working on simplifying it further.

Reply:
New version fixes the half spaces for even and odd, and repairs the symmetry.  Noether strikes again!